src/studio/src/designer/frontend/app-development/features/dataModeling/containers/DataModelingContainer.tsx

```diff
@@ -61,6 +62,8 @@ export default function DataModelingContainer(props: IDataModelingContainerProps
     fetchModel(schema.id);
   };
   const onSaveSchema = (schema: any) => {
+    // eslint-disable-next-line no-param-reassign
+    schema.$id = getStaticDataModelUrl(dataModelName);
```

Comment:
Why are you reassigning instead of creating a save copy here?
```javascript
const onSaveSchema = (prevSchema: any) => {
  const schema = {...prevSchema, '$id' = getStaticDataModelUrl(dataModelName), };
  dispatch(saveDataModel({ schema }));
}
```

Reply:
Because it looks cleaner, do you really want it that way?

Reply:
@lorang92 @nkylstad what do you think

Reply:
fixed